test/pkgcheck.sh

```diff
@@ -40,24 +100,59 @@ cd btmp1
   Darwin)
     export LDFLAGS="-Wl,-headerpad_max_install_names"
     ;;
-  Linux)
-    if grep -i fedora /etc/os-release > /dev/null
-    then
-        # Note: Fedora patches cmake to use -O2 in release, which
-        # does not match the -O3 configure sets :-(
-        export CFLAGS="-O2 -DNDEBUG"
-    fi
-    ;;
   esac
-  ../configure
+  # Use same optimization level as cmake did.
+  CFLAGS="$(awk -F= '/CMAKE_C_FLAGS_RELEASE:STRING=/ {print $2}' < ../btmp2/CMakeCache.txt)"
+  export CFLAGS
+  # Hack: oddly, given CHOST=powerpc-linux-gnu, configure concludes uname is gnu,
+  # causing it to set LDSHAREDFLAGS to a different value than cmake uses.
+  # So override it when appropriate.  FIXME
```

Comment:
Perhaps fix this powerpc-detection in configure instead?

Reply:
I believe the issue is on `configure:26`:
https://github.com/zlib-ng/zlib-ng/blob/9abe04b926ef688efa3d4f464209246a22afb01d/configure#L25-L28
When `CHOST` is something like `powerpc-linux-gnu`, it will match the second pattern, which will select the last string after the `-`. But this issue happens with any `CHOST` value in the form `first-second-third`, such as `arm-linux-gnueabihf` and `aarch64-linux-gnu`. All these match the second pattern.

That change may deserve a separate PR though. For now I believe setting `--uname=linux` will suffice.

Reply:
Just opened a new issue about this so it doesn't get forgotten: https://github.com/zlib-ng/zlib-ng/issues/666

Reply:
I added a commit to address this.